File: EngTaskGradingNetBESolution/EngTaskGradingNetBE/Models/Dtos/GradeDto.cs
namespace EngTaskGradingNetBE.Models.Dtos
{
  public record GradeCreateDto(int TaskId, int StudentId, int Value, string? Comment);
  public record GradeUpdateDto(int Value, string? Comment);
  public record GradeDto(int Id, int TaskId, int StudentId, int Value, DateTime Date, string? Comment);
  public record GradeSetDto(List<TaskDto> Tasks, List<StudentDto> Students, List<GradeDto> Grades);


  public record NewGradeSetDto(List<NewGradeSetTaskDto> Tasks);
  public record NewGradeSetStudentDto(StudentDto Student, List<GradeDto> Grades);
  public record NewGradeSetTaskDto(TaskDto Task, List<NewGradeSetStudentDto> Students);
}

File: eng-task-grading-react-fe/src/model/grade-dto.ts
// --- NewGradeSet DTOs ---
import type { TaskDto } from "./task-dto";
import type { StudentDto } from "./student-dto";

export interface NewGradeSetStudentDto {
  student: StudentDto;
  finalValue: number | null;
  grades: GradeDto[];
}

export interface NewGradeSetTaskDto {
  task: TaskDto;
  students: NewGradeSetStudentDto[];
}

export interface NewGradeSetDto {
  tasks: NewGradeSetTaskDto[];
}

export interface GradeDto {
  id: number;
  taskId: number;
  studentId: number;
  value: number;
  comment: string | null;
  date: Date;
}

export interface GradeCreateDto {
  taskId: number;
  studentId: number;
  value: number;
  comment: string | null;
}

export interface GradeUpdateDto {
  value?: number;
  comment?: string | null;
}

export interface GradeSet {
  tasks: TaskDto[];
  students: StudentDto[];
  grades: GradeDto[];
}

